apps/dot_ext/utils.py

```diff
@@ -34,3 +41,25 @@ def remove_application_user_pair_tokens_data_access(application, user):
         refresh_token_delete_cnt = RefreshToken.objects.filter(application=application, user=user).delete()[0]
 
     return data_access_grant_delete_cnt, access_token_delete_cnt, refresh_token_delete_cnt
+
+
+def validate_app_is_active(request):
+
+    client_id = None
+
+    if request.GET is not None:
+        client_id = request.GET.get('client_id', None)
+
+    if client_id is None and request.POST is not None:
+        client_id = request.POST.get('client_id', None)
+
+    app = None
+
+    if client_id is not None:
+        try:
+            app = Application.objects.get(client_id=client_id)
+        except Application.DoesNotExist:
+            pass
+
+    if app and not app.active:
+        raise PermissionDenied(settings.APPLICATION_TEMPORARILY_INACTIVE.format(app.name))
```

Comment:
Stylistic choice but this could be moved inside the try statement since this condition can only be satisfied if `app` returns a valid result.